core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -711,7 +709,7 @@ protected DataReader initDataReader(DataReader reader) {
                 setElevationProvider(eleProvider).
                 setWorkerThreads(dataReaderWorkerThreads).
                 setWayPointMaxDistance(dataReaderWayPointMaxDistance).
-                setWayPointElevationMaxDistance(elevationWayPointMaxDistance).
```

Comment:
We could also make elevation way point max distance controllable through a request parameter for routing?

Reply:
Thanks a lot for this very nice PR @msbarry! 

As far as I can tell the `elevationWayPointMaxDistance` is used to control how many extra pillar nodes get created (to preserve elevation changes of the geometries) and it is also used during path simplification when running routing requests (to not throw away way points needed to reflect elevation changes). These two usages are different in that the first only can have an effect during import (after the first import it becomes meaningless unless we rebuild the graph) while the other could (at least in theory) be changed for every single request. Is that correct? For this reason I think it would be cleaner to use two parameters, e.g. one would go into the (not yet existing) `ImporterConfig` and the other would go to `RouterConfig` and the latter could also be changed via the routing hints in `Router#createPathMerger`. I just wonder if this is overkill and I do not think its urgent to separate this into two parameters already. In any case the parameter used for the import is a good default value for the one used during routing?

> Should we still keep the two variables?
Which two variables do you mean? I thought so far there is only one (only one that has to do with import and routing).

> We could also make elevation way point max distance controllable through a request parameter for routing?

Yes, exactly, in `Router#createPathMerger`.

Reply:
Thanks @easbar, longEdgeSamplingDistance controls how many extra pillar nodes are added and elevationWayPointMaxDistance prevents them from getting removed during simplification (on import, and during routing).  I added a new elevation_way_point_max_distance request parameter in https://github.com/graphhopper/graphhopper/pull/1953/commits/9edbfbf2002b3309ef816ed7ac6777aeaa5d5028 to let you change elevationWayPointMaxDistance on each request or omit to default to what was used during import.  I agree it's probably overkill to split out separate config parameters for data reader import and router default. Let me know if that change looks sufficient.

Reply:
Yes thanks I like it.